variables.tf

```
@@ -221,3 +221,14 @@ variable "availability_set_id" {
   type        = string
   default     = null
 }
+
+variable "license_type" {
+  description = "Specifies the type of license which should be used for this Virtual Machine. Value must be \"None\", \"Windows_Client\", \"Windows_Server\", \"RHEL_BYOS\" or \"SLES_BYOS\". Defaults to null."
+  type        = string
+  default     = null
+
+  validation {
+    condition     = contains(["None", "Windows_Client", "Windows_Server", "RHEL_BYOS", "SLES_BYOS"], var.license_type)
+    error_message = "License mode must be \"None\", \"Windows_Client\", \"Windows_Server\", \"RHEL_BYOS\" or \"SLES_BYOS\"."
```

@stiross79 Try something like this:

```suggestion
    condition     = contains(["None", "Windows_Client", "Windows_Server", "RHEL_BYOS", "SLES_BYOS", null], var.license_type)
    error_message = "License mode must be \"None\", \"Windows_Client\", \"Windows_Server\", \"RHEL_BYOS\", \"SLES_BYOS\" or null."
```

You'll have to do something similar for the resource preconditions.